build/workflow/Makefile

```diff
@@ -81,6 +83,27 @@ ${TESTS}: setup-testing
 >@${MESSAGE} end "Passed." "Failed."
 >@rm -f status_code
 
+${STANDALONE_MODULE_TEST}: ${MODULE_TESTS}
+>@test_target=$$(echo $@ | sed 's/..\/..\/${TEST_LOCATION}\/${MODULE_NAME}\///g') ; \
+    test_script=$${test_target/test-/} ; \
```

Comment:
I copied this from the `${TESTS}` rule but I don't understand it. I don't think we have any test files that start with `test-` instead of `test_`?

Reply:
The line
```make
MODULE_TESTS := $(foreach test_file,$(MODULE_TEST_FILES),test-$(test_file))
```
prepends `test-` in order to create a **target** name. `make` targets are not the same as files, and in this case it is especially important that the test file not be confused with the test target. The reason is that `make` will specially interpret target names which coincide with file names, as the names of files that the recipe portion (the script) is supposed to create.

The lines you are referring to just recover the test file name from the target name. It's roundabout and ugly because I'm not an expert in make string processing.

Reply:
If you'd like, in your case you could hard-code a single target and recipe, removing the need for the string name processing.

Reply:
Like:
```make
standalone-module-test-1: setup-testing
>@bash standalone_module_tests/my_special_test.sh; \
    echo "$$?" > status_code;
...
```

then add to the definition of the test list:

```make
TESTS := ${UNIT_TESTS} ${MODULE_TESTS} standalone-module-test-1
```